config/routes.rb

```diff
@@ -1,6 +1,10 @@
-Hogwarts::Application.routes.draw do
+Rails.application.routes.draw do
+  # get '/houses/:id' => '/#show'
 
-  root "house#index"
+# Hogwarts::Application.routes.draw do
+  get 'houses/:name' => 'houses#show'
+  get 'houses/index'
+  root 'houses#index'
   resource :house, only: [:index, :show]
```

Comment:
You're missing an "s" at the end of `resources`. `resource` (singular) does have a purpose, but it's not what you want here.